### ISSUE TITLE
Many missing Arabic positional glyphs

### ISSUE THREAD

**Issue by dscorbett** [notofonts/arabic#10]
### Font
NotoKufiArabic-Regular.otf
NotoNaskhArabic-Regular.otf
NotoNastaliqUrdu-Regular.otf
NotoSansArabic-Regular.otf

### Where the font came from, and when
Site: https://github.com/googlefonts/noto-fonts/blob/97e5bf0bbaf85c14498696b408d60b517b0e9e70/unhinted/otf/NotoKufiArabic/NotoKufiArabic-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/97e5bf0bbaf85c14498696b408d60b517b0e9e70/unhinted/otf/NotoNaskhArabic/NotoNaskhArabic-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/97e5bf0bbaf85c14498696b408d60b517b0e9e70/unhinted/otf/NotoNastaliqUrdu/NotoNastaliqUrdu-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/97e5bf0bbaf85c14498696b408d60b517b0e9e70/unhinted/otf/NotoSansArabic/NotoSansArabic-Regular.otf
Date: 2022-02-08

### Font version
Noto Kufi Arabic: Version 2.106
Noto Naskh Arabic: Version 2.012
Noto Nastaliq Urdu: Version 3.002
Noto Sans Arabic: Version 2.009

### Issue
Some of the Arabic fonts are missing some positional forms. A few have been reported in notofonts/noto-fonts#673, notofonts/arabic#75, and notofonts/arabic#69, but it is easier to find them all at once by running a script. I encourage you to write an analogous Font Bakery check.
```bash
#!/bin/bash

if [ $# -ne 2 ]
then
    echo "Usage: $0 ArabicShaping.txt FONT_DIRECTORY"
    exit 1
fi

shaping_file="$1"
font_dir="$2"

check () {
    local cp="$1"
    local position="$2"
    local dflt="$3"
    local positional="$4"
    local font="$5"
    if [ "$dflt" = "$positional" ]
    then
        printf "$cp\t$position\t$font\n"
    fi
}

check-positions () {
    local jt="$1"
    grep "^[^#].*; $jt; " "$shaping_file" \
    | cut -f 1 -d ';' \
    | while read cp
    do
        for font in "$font_dir"/Noto{{Kufi,Naskh,Sans}Arabic,NastaliqUrdu}-Regular.?tf
        do
            local dflt="$(hb-shape --ned --no-positions --remove-default-ignorables "$font" -u $cp)"
            if [[ $dflt = *notdef* ]]
            then
                continue
            fi
            local init
            if [ $jt = L -o $jt = D ]
            then
                init="$(hb-shape --ned --no-positions --remove-default-ignorables "$font" -u "$cp 200D")"
            fi
            check $cp init "$dflt" "$init" "$(basename "$font")"
            local medi
            if [ $jt = D ]
            then
                medi="$(hb-shape --ned --no-positions --remove-default-ignorables "$font" -u "200D $cp 200D")"
            fi
            check $cp medi "$dflt" "$medi" "$(basename "$font")"
            local fina
            if [ $jt = D -o $jt = R ]
            then
                fina="$(hb-shape --ned --no-positions --remove-default-ignorables "$font" -u "200D $cp")"
            fi
            check $cp fina "$dflt" "$fina" "$(basename "$font")"
        done
    done
}

for jt in L D R
do
    check-positions $jt
done
```

Here are the results for the current versions.
```
0620	init	NotoKufiArabic-Regular.otf
0620	medi	NotoKufiArabic-Regular.otf
0620	fina	NotoKufiArabic-Regular.otf
063B	init	NotoKufiArabic-Regular.otf
063B	medi	NotoKufiArabic-Regular.otf
063B	fina	NotoKufiArabic-Regular.otf
063C	init	NotoKufiArabic-Regular.otf
063C	medi	NotoKufiArabic-Regular.otf
063C	fina	NotoKufiArabic-Regular.otf
063D	init	NotoKufiArabic-Regular.otf
063D	medi	NotoKufiArabic-Regular.otf
063D	fina	NotoKufiArabic-Regular.otf
063E	init	NotoKufiArabic-Regular.otf
063E	medi	NotoKufiArabic-Regular.otf
063E	fina	NotoKufiArabic-Regular.otf
063F	init	NotoKufiArabic-Regular.otf
063F	medi	NotoKufiArabic-Regular.otf
063F	fina	NotoKufiArabic-Regular.otf
077A	init	NotoKufiArabic-Regular.otf
077A	medi	NotoKufiArabic-Regular.otf
077B	init	NotoKufiArabic-Regular.otf
077B	medi	NotoKufiArabic-Regular.otf
08AC	fina	NotoNaskhArabic-Regular.otf
08AC	fina	NotoSansArabic-Regular.otf
08B1	fina	NotoKufiArabic-Regular.otf
08B1	fina	NotoNaskhArabic-Regular.otf
08B1	fina	NotoSansArabic-Regular.otf
```

**Comment by visionario2** [comment]
So have all the fixes that were included in version 2.5 on the old page been rolled into version 2.010 here?

**Comment by khaledhosny** [comment]
With #204, running the script gives:

```
0886	init	NotoKufiArabic[wght].ttf
0886	fina	NotoKufiArabic[wght].ttf
088E	fina	NotoKufiArabic[wght].ttf
08B1	fina	NotoKufiArabic[wght].ttf
0886	init	NotoNaskhArabic[wght].ttf
0886	fina	NotoNaskhArabic[wght].ttf
088E	fina	NotoNaskhArabic[wght].ttf
08B1	fina	NotoNaskhArabic[wght].ttf
0886	init	NotoSansArabic[wdth,wght].ttf
0886	fina	NotoSansArabic[wdth,wght].ttf
088E	fina	NotoSansArabic[wdth,wght].ttf
08B1	fina	NotoSansArabic[wdth,wght].ttf
```

These code points don’t have attestations for are positional forms, so the representation glyph is already a final form etc.
